posts/what-makes-entando-a-highly-avialable-architecture.md

```diff
@@ -0,0 +1,81 @@
+
+### Durability
+
+Entando has options to configure a shared cache across multiple nodes. This configuration lends to availability in that if one of the nodes crashes at an application or hardware level, the data you were working on will still be available. It is also possible to configure Entando to use Redis as a shared cache or data store.
+
+# Databases
+
+Out of the box, Entando will create databases inside the Kubernetes cluster. Each application will have a database and mechanism by which it can import and export data. An external database will set up a more scalable approach. Entando makes these a breeze to set up by using dedicated custom resource definitions. Customers will have to configure the external database themselves, but once complete, adding it to the custom resource definition is very straightforward. The Entando Operator will automatically create tables, indices, and foreign keys.
+
+Scaling a database used with Entando is very similar to scaling databases in general.
+
+### Horizontal Scaling
+
+Database scaling can happen in one of two different ways. The first way has already been mentioned and is the exact mechanism for scaling an application server vertically. Once you have reached the point of maximum vertical scalability, it's time to start scaling horizontally. The horizontal scaling of a database includes both reads and writes.
+
+### Scaling Database Reads
+
+A database read happens when a user or program requests data stored in the database. The quickest and easiest way to scale reads is to increase the number of database replicas and then split reads and writes from there. Depending on the database, it will employ a proxy or database driver for functionality.
+
+Caching techniques can also help scale database reads. Storing frequently accessed data in a separate cache keeps the load off of the database.  When a user or an application goes to access the data, it may not even hit the database, pulling from the cache instead.
+
+A third way to scale database reads is to implement database sharding. This strategy is also the primary method used to scale database writes.
+
+### Scaling Database Writes
+
+A database write happens when a user or program wishes to store a piece of data on the database. Since every database replica will also need to contain this piece of data, increasing replicas does not fix the issue as it does for database reads. Enter what is known as "database sharding."
+
+Sharding a database is the process of spreading the data across multiple databases. Spreading the data out can be as simple as putting records that start with the character 'A' through 'L' in one database and the remaining 'M' to 'Z' records in a different database. This arrangement means we need another component in front of the database to handle the routing. Some databases support sharding natively, and others will require a third-party tool.
```

Comment:
Is this term coined by Entando - or is this an often-used term across tech?

Reply:
Sharding is a well-known term in Database land.